### PR TITLE
Com.GitHub.kalibari.audok

### PR DESCRIPTION
**Pull request by kalibari**
### Please confirm your submission meets all the criteria

- [ x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x ] My pull request follows the instructions at [App Submission][submission].
- [x ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [ x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
Since this is a duplicate of #2416, can close it?

(it wasn't necessary to open a new one)

File: com.github.kalibari.audok.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+              "type": "archive",
+              "url": "https://github.com/ytdl-org/youtube-dl/releases/download/2020.02.16/youtube-dl-2020.02.16.tar.gz",
+              "sha256": "abcd19c109125a69915c6ad70fcc6b02411a64427fbb7b91a6b37217dfb9e0c6"
+            }],
+            "post-install": ["install -Dm0755 -t /app/bin youtube-dl"]
+        },
+        {
+            "name" : "audok",
+            "no-autogen": true,
+            "no-make-install": true,
```
Comment (hfiguiere) [inline com.github.kalibari.audok.json:63]: This won't install the binaries, or did I miss something?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build com.github.kalibari.audok

**Comment by hfiguiere** [comment]
x86_64 fail to build because: `• file-invalid          : <screenshot> failed to load [https://github.com/kalibari/audok/blob/master/audok/screenshot.png]`

aarch64: streamripper needs to have the configure files updated.

File: flatpak.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flatpak.json:2]: why????

File: com.github.kalibari.audok.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            "sources": [{
+              "type": "archive",
+              "url": "https://github.com/ytdl-org/youtube-dl/releases/download/2020.02.16/youtube-dl-2020.02.16.tar.gz",
+              "sha256": "abcd19c109125a69915c6ad70fcc6b02411a64427fbb7b91a6b37217dfb9e0c6"
+            }]
+        },
+        {
+            "name" : "audok",
+            "no-autogen": true,
+            "no-make-install": false,
```
Comment (hfiguiere) [inline com.github.kalibari.audok.json:68]:
```suggestion
```

File: com.github.kalibari.audok.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            "sources": [{
+              "type": "archive",
+              "url": "https://downloads.sourceforge.net/sourceforge/streamripper/streamripper-1.64.6.tar.gz",
+              "sha512": "8477086d0099f6de861f1c47112476c427073b6e8127bbaaa2dd3e7930fe9a5c6d9b2c68d4a317f1bc0b20c625e665d5c245189049d35468aa83cf51828d4428"
+            }]
+        },
+        {
+            "name": "youtube-dl",
+            "no-autogen": true,
+            "no-make-install": false,
```
Comment (hfiguiere) [inline com.github.kalibari.audok.json:57]:
```suggestion
```

File: com.github.kalibari.audok.json [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            "make-install-args": ["PREFIX=/app"],
```
Comment (hfiguiere) [inline com.github.kalibari.audok.json:58]:
```suggestion
```

You shouldn't need this.

File: com.github.kalibari.audok.json [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+              "type": "archive",
+              "url": "https://www.ffmpeg.org/releases/ffmpeg-4.3.2.tar.xz",
+              "sha256": "46e4e64f1dd0233cbc0934b9f1c0da676008cad34725113fb7f802cfa84ccddb"
+            }],
+            "cleanup": ["/share/ffmpeg/examples"]
+        },
+        {
+            "name": "streamripper",
+            "no-autogen": true,
+            "no-make-install": false,
```
Comment (hfiguiere) [inline com.github.kalibari.audok.json:44]:
```suggestion
```

**Comment by kalibari** [comment]
> Since this is a duplicate of #2416, can close it?
> 
> (it wasn't necessary to open a new one)

Hello  hfiguiere, yes please close it, thank you!

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
> > Since this is a duplicate of #2416, can close it?
> > (it wasn't necessary to open a new one)
> 
> Hello hfiguiere, yes please close it, thank you!

I can't but you can and should.

**Comment by hfiguiere** [comment]
bot, build com.github.kalibari.audok

**Comment by barthalion** [comment]
Building streamripper fails on aarch64. You can fix it like this: https://github.com/flathub/org.gimp.GIMP/blob/f819d5614ca19d3ffff99f2040881a873daa175f/org.gimp.GIMP.json#L249

File: flatpak.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (hfiguiere) [inline flatpak.json:1]: This file does nothing anyway as it has the wrong filename.

See https://github.com/flathub/flathub/pull/2420#issuecomment-879684329 on how to resolve the aarch64 build issue.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Also none of the suggestion I made were applied albeit they were resolved.

**Comment by nedrichards** [comment]
bot, build com.github.kalibari.audok

File: com.github.kalibari.audok.json [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+            },
+            "sources": [
+                {
+                "type": "archive",
+                "url": "https://downloads.sourceforge.net/sourceforge/streamripper/streamripper-1.64.6.tar.gz",
+                "sha512": "8477086d0099f6de861f1c47112476c427073b6e8127bbaaa2dd3e7930fe9a5c6d9b2c68d4a317f1bc0b20c625e665d5c245189049d35468aa83cf51828d4428"
+                },
+                {
+                    "type": "shell",
+                    "commands": [ "cp -p /usr/share/automake-*/config.{sub,guess} ." ]
```
Comment (hfiguiere) [inline com.github.kalibari.audok.json:56]:
```suggestion
                    "commands": [ 
                        "cp -p /usr/share/automake-*/config.{sub,guess} .",
                        "cp -p /usr/share/automake-*/config.{sub,guess} libmad-0.15.1b/"
                    ]
```

This should fix the build. There is a subproject in the tarball that needed the same treatment.

File: com.github.kalibari.audok.json [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+            "sources": [{
+              "type": "archive",
+              "url": "https://github.com/ytdl-org/youtube-dl/releases/download/2020.02.16/youtube-dl-2020.02.16.tar.gz",
+              "sha256": "abcd19c109125a69915c6ad70fcc6b02411a64427fbb7b91a6b37217dfb9e0c6"
+            }]
+        },
+        {
+            "name" : "audok",
+            "no-autogen": true,
+            "no-make-install": false,
```
Comment (hfiguiere) [inline com.github.kalibari.audok.json:74]:
```suggestion
```

You don't need that line.

**Comment by kalibari** [comment]
update com.github.kalibari.audok.json

**Comment by kalibari** [comment]
audo

**Comment by hfiguiere** [comment]
bot, build com.github.kalibari.audok

**Comment by hfiguiere** [comment]
bot, build com.github.kalibari.audok

**Pull request closed without merging**